src/extension/extension.ts

```diff
@@ -127,11 +128,13 @@ export class RunmeExtension {
     )
 
     const reporter = new GrpcReporter(context, server)
-    const serializer = grpcSerializer
-      ? new GrpcSerializer(context, server, kernel)
-      : new WasmSerializer(context, kernel)
+    const serializer = useConnectSerializer
```

Comment:
Using the "?" : with nested "?": isn't very readable.
I think an if/else would be more readable.

Per the comment above the argument for the connectSerializer option should be a string containing the address. You should use non-empty string to indicate that connectSerializer should be used. Otherwise you should fall back to the existing logic.

Reply:
resolved in latest verison.